### DISCUSSION TITLE
Support for snapshot / restore for different Cloud-hypervisor version

### DISCUSSION THREAD

**Discussion by praveenkumr** [cloud-hypervisor/cloud-hypervisor#4950] [Q&A]
Hello All,

As per the https://github.com/cloud-hypervisor/cloud-hypervisor#3-status 

"Snapshot/restore is not supported across different versions"

Wanted to check if this still a valid statement or we support Snapshot / restore with different Cloud-hypervisor versions?

**Comment by rbradford** [comment]
Thanks @praveenkumr. I think we can adjust that now! We definitely support live migration from v28.0 to v28.x (LTS releases) and also currently from v26.0 to v28.0.

With the work that @sboeuf is doing to improve our restore logic I think that that might break from v29.0.

**Reply by sboeuf** [reply]
Well so far I didn't modify any of the `state` structures, meaning I didn't break compatibility. The compatibility will be broken when I'll try to improve the existing state structures after I'm done with the global rearrangement of the code.

**Reply by praveenkumr** [reply]
@sboeuf is there a way that backward compatibility is maintained, i.e., we can restore guest vms using say latest v29.0 Cloud-hypervisor (CH) binary from the snapshot created by v26.0 (CH) binary?

**Reply by rbradford** [reply]
With versionize if you have fields you no longer need you can just ignore them; this would be require filling out the conversion code for each struct.
